src/ejabberd_sm.erl

```diff
@@ -1077,3 +1095,48 @@ get_cached_unique_count() ->
 -spec sm_backend() -> backend().
 sm_backend() ->
     ejabberd_sm_backend:backend().
+
+%%====================================================================
+%% Deprecated API
+%%====================================================================
+open_session(SID, U, S, R, Info) ->
```

Comment:
As pointed out by @erszcz these functions can be marked as `-deprecated`. More info can be found in `xref` doc: http://erlang.org/doc/man/xref.html. 

Also I'm wondering if maybe module like `mongoose_deprecations` could be used to produce warning logs when such function is used.